### ISSUE TITLE
vue/no-bare-strings-in-template: allowlist ignores "&ndash;"

### ISSUE THREAD

**Issue by ilyub** [vuejs/eslint-plugin-vue#1739]
**Tell us about your environment**

- **ESLint version:** 8.4.1
- **eslint-plugin-vue version:** 8.2.0
- **Node version:** 14.18.0
- **Operating System:** Windows 10

**Please show your full configuration:**
```ts
rules: {
    "vue/no-bare-strings-in-template": [
      "warn",
      {
        allowlist: [
          "&ndash;"
        ],
```

**What did you do?**	

```vue
<div>&ndash;</div>
```

**What did you expect to happen?**

No warning since &ndash; is added to allowlist

**What actually happened?**

Getting warning

**Repository to reproduce this issue**

https://github.com/ilyub/repro-no-bare-strings-in-template/blob/develop/src/components/IconPicker.vue

**Comment by ota-meshi** [comment]
Thank you for this issue.

Rule settings do not understand HTML entities. You need to specify in the characters that are actually displayed.

```js
    "vue/no-bare-strings-in-template": [
      "warn",
      {
        allowlist: [
          "\u2013"
        ],
      }
    ]
```
